infrasec/tutorials/atlantis.md

```diff
@@ -0,0 +1,418 @@
+    }
+    ```
+
+    Once we've merged the PR, we'll have the cert validation output value to validate the certificate in the next step.
+
+3. Validate the Route53 DNS records in the Commercial account using the [`aws_acm_certificate_validation` resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acm_certificate_validation).
+
+    We'll need to find the `arn` to plug into the [atlantis module](https://registry.terraform.io/modules/terraform-aws-modules/atlantis/aws/latest) we'll call in the next step. We can find this in the console's Certificate Manager after merging the PR to add the certificate.
+
+    <img src="https://github.com/trussworks/Engineering-Playbook/blob/3efe6ea02ed010f3db2c07921c5c8acc60406b84/infrasec/tutorials/images/atlantis_acm1.png" width="450">
+
+### Set up Docker 🐋
+
+For any project not requiring a specific type of docker implementation, we can simply [pass in the `trussworks-atlantis-ecs-image` image](https://github.com/trussworks/trussworks-atlantis-ecs-image) when we make the module call in [Step 3](#step-3-call-the-atlantis-module-).
+
+If we do not set the `atlantis_image` variable, we'll find `atlantis:latest` is used by default. This default is not recognized as updated when a new "latest" is released due to the word remaining unchanged. Therefore, we recommend passing in a numbered version of the Atlantis docker image to the `atlantis_image` var.
+
+Note that the use of Docker in tandem with Atlantis is actually optional, however both our example usage here and the Atlantis Fargate module expect it.
+
+## Step 3: Call the [Atlantis module](https://registry.terraform.io/modules/terraform-aws-modules/atlantis/aws/latest) 🧜
+
+We'll start with the most basic Atlantis module call we can. Assuming an existing VPC, Robot user, validated certificate, zone name, Atlantis Docker image, and GitHub connection, the code will look something like this
+
+  ```hcl
+  module "atlantis" {
+    source  = "terraform-aws-modules/atlantis/aws"
+    version = "~> 2.36.0" // make sure to update to the latest version
+
+    name = "${var.environment}-atlantis"
+
+    atlantis_image = "milmove/circleci-docker:milmove-atlantis"
+
+    # Route53
+    create_route53_record = false
+    route53_zone_name = "atlantis.exp.move.mil"
+    certificate_arn = "arn:aws-us-gov:acm:us-gov-west-1:015681133840:certificate/456c11bd-a730-476d-ba7c-1bd4690ede3f"
+
+    # VPC
+    vpc_id          = ""
+    cidr            = "10.30.0.0/16"
+    azs             = ["us-gov-east-1a", "us-gov-east-1b", "us-gov-east-1c"]
+    private_subnet_ids = ["subnet-012345678f01", "subnet-012345678f02", "subnet-012345678f03"]
+    public_subnet_ids  = ["subnet-012345678f04", "subnet-012345678f05", "subnet-012345678f06"]
+
+    # Atlantis
+    atlantis_github_user = "atlantis"
+    atlantis_github_user_token = data.aws_ssm_parameter.github_user_token.value
```

Comment:
```suggestion
```
I know we configured it this way in legendary-waddle, but this can also be configured directly in the `custom_environment_secrets` section, which avoids storing the SSM parameter in the state file and also avoids a lot of the complicated logic in the atlantis terraform module around these settings.

For example, the `atlantis_github_user_token` could be deleted and replaced with something like this in `custom_environment_secrets`: 

```    
   {
      name      = "ATLANTIS_GH_TOKEN"
      valueFrom = local.ssm_github_token_name
    },
```